src/managed_process.rs

```diff
@@ -16,17 +16,67 @@ use std::time::{ SystemTime, Duration, UNIX_EPOCH };
 #[derive(PartialEq, Eq, Clone, Copy, Debug)]
 pub struct ExitStatus(c_int);
 
-const RESTART_TIME_THRESHOLD: f64 = 5.0; // seconds
+const RESTART_TIME_THRESHOLD: u64 = 5; // seconds
 
-fn seconds_since_epoch() -> f64 {
+fn duration_since_epoch() -> Duration {
     let now = SystemTime::now();
-    now.duration_since(UNIX_EPOCH).unwrap().as_secs() as f64
+    now.duration_since(UNIX_EPOCH).unwrap()
 }
 
 pub struct ManagedProcess {
     kill_signal: Arc<Mutex<u32>>,
     pid:         Arc<Mutex<Option<u32>>>,
-    thread:      JoinHandle<()>
+    thread:      JoinHandle<()>,
+    backoff:     Arc<RwLock<Backoff>>,
+}
+
+struct Backoff {
+    restart_count: u64,
+    restart_threshold: Duration,
+    start_time: Duration,
+    end_time: Duration,
```

Comment:
I think `start_time` and `end_time` should be `std::time::Instant`, not `Duration`.